Code cleanup

critias opened the pull request:
I removed a bunch of old unused code and imports together with a some other minor cleanup.

In addition I marked a few helper functions of Path as deprecated. I would rather see this functionality inside the recipes than inside of sisyphus. The fuse filesystem is also marked as deprecated since I haven't used or tested it in a long time and don't know if anybody else is still using it. I hope to get some feedback this way if it's still used somewhere.

Any opinions about removing this functionally in the long run?

michelwi commented:
The features you mention are not used in our recipes afaik. And I agree on deprecating the Path helper functions to be more restrictive on accessing Path objects in the manager.

michelwi commented:
As some of the cleanup involved unused imports and code formatting, maybe it would make sense to add (or augment) some of the code (style) checkers in the test pipeline?

I see some flake warnings about whitespaces... 
Other repos make use of black for code formatting. maybe this could be an option?

mattiadg commented on sisyphus/manager.py:
```diff
@@ -63,6 +64,8 @@ def manager(args):
     # try to load fuse filesystem
     filesystem = None
     if args.filesystem:
+        warnings.warn("The filesystem is planed to be removed. Let the authors know if you still need it!",
```
typo in planed

JackTemaki replied:
```suggestion
        warnings.warn("The filesystem is planned to be removed. Let the authors know if you still need it!",
```

mattiadg commented:
Continuing on @michelwi's comment, [ruff](https://github.com/charliermarsh/ruff) is an extremely fast linter with a lot of options. It is for sure a great support for this kind of operations

critias commented:
> I see some flake warnings about whitespaces... Other repos make use of black for code formatting. maybe this could be an option?

flake8 didn't complain in my case, it is setup to run automatically before each commit using .pre-commit-config.yaml. Are you using the same settings as set in .flake8?
I wouldn't  mind running running black on the whole code base once, but would prefer using a linter afterwards.

@mattiadg what are the advantages of ruff here compared to flake8? The code base is small enough right now that speed wasn't really an issue for me.

michelwi commented:
> > I see some flake warnings about whitespaces... Other repos make use of black for code formatting. maybe this could be an option?
> 
> flake8 didn't complain in my case, it is setup to run automatically before each commit using .pre-commit-config.yaml. Are you using the same settings as set in .flake8?

I am on the Github page, clicking on "Show all checks" -> "Details" (of any of the tests), then open the passed test "lint with flake8". there are a couple of warnings like
```
./sisyphus/tools.py:67:15: E275 missing whitespace after keyword
./sisyphus/delayed_ops.py:197:1: W391 blank line at end of file
./sisyphus/visualize.py:76:63: E741 ambiguous variable name 'l'
```
This is what I was referring to.

critias commented:
> This is what I was referring to.

Thanks for pointing it out. Most errors where just () around an assertion. I fixed all issues and changed the workflow settings to fail the build if flake8 complains.
